refgenconf/refgenconf.py

```diff
@@ -2087,11 +2164,148 @@ def _select_genomes(self, genome=None, strict=False, order=None,
+                            for i in range(len(asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_CHILDREN_KEY])):
+                                asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_CHILDREN_KEY][i] = asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_CHILDREN_KEY][i].replace(
+                                    genome, digest)
+                        elif asset != "fasta":
+                            for i in range(len(asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_PARENTS_KEY])):
+                                asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_PARENTS_KEY][i] = asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_PARENTS_KEY][i].replace(
+                                    genome, digest)
+
+                # use the genome digest as primary keys
+                rgc[CFG_GENOMES_KEY][digest] = rgc[CFG_GENOMES_KEY].pop(
+                    genome)
+                # remove old "genome_digest" section
+                del rgc[CFG_GENOMES_KEY][digest][CFG_CHECKSUM_KEY]
+            else:
+                missing_digest.append(genome)
+                del rgc[CFG_GENOMES_KEY][genome]
+
+        return missing_digest
+
+    # restructure the genome_folder
+
+    def alter_file_tree():
+        my_genome = {}
+        for k, v in rgc[CFG_GENOMES_KEY].items():
+            my_genome.update([(v[CFG_ALIASES_KEY][0], k)])
+
+        os.mkdir(os.path.abspath(os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR)))
+        os.mkdir(os.path.abspath(os.path.join(rgc[CFG_FOLDER_KEY], ALIAS_DIR)))
+
+        # copy folder
+        for root, dirs, files in os.walk(rgc[CFG_FOLDER_KEY]):
+            for dir in dirs:
+                if dir in my_genome:
+                    shutil.copytree(os.path.join(rgc[CFG_FOLDER_KEY], dir),
+                                    os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR, dir), ignore=ignore_patterns("*.DS_Store"))
+            del dirs[:]
+
+        for root, dirs, files in os.walk(os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR)):
+            for dir in dirs:
+                _swap_names_in_tree(os.path.join(
+                    root, dir), my_genome[dir], dir)
+                os.mkdir(os.path.join(rgc[CFG_FOLDER_KEY], ALIAS_DIR, dir))
+                # create symlink for alias folder
+                for genome, assets, files in os.walk(os.path.join(root, my_genome[dir])):
+                    for asset in assets:
+                        old_path = os.path.join(genome, asset)
+                        new_path = old_path.replace(
+                            my_genome[dir], dir).replace("/data/", "/alias/")
+                        os.mkdir(new_path)
+                    for file in files:
+                        old_path = os.path.join(genome, file)
+                        new_path = old_path.replace(
+                            my_genome[dir], dir).replace("/data/", "/alias/")
+                        os.symlink(old_path, new_path)
```

Comment:
don't hardcode `os.symlink`, make this function an argument so the type of function making the link is configurable. Maybe someone would like to use `copy` for some reason instead